backend/posts/urls.py

```diff
@@ -1,13 +1,15 @@
 from django.urls import path, include
+from django.conf.urls import url
 from rest_framework.routers import DefaultRouter
-from . import views
+from .views import posts, votes
 
 router = DefaultRouter()
-router.register('posts', views.PostViewSet)
-router.register('users', views.UserViewSet)
-router.register('comments', views.CommentViewSet)
-router.register('votes', views.VoteViewSet)
+router.register('posts', posts.PostViewSet)
+router.register('users', posts.UserViewSet)
+router.register('comments', posts.CommentViewSet)
 
 urlpatterns = [
     path('', include(router.urls)),
+    url(r'^posts/(?P<post_pk>\d+)/(?P<vote_type>up|down)?votes/$', votes.post_votes),
```

Comment:
If you like to use regex this approach is fine, but you can try also the same with `path`, should be more easier to read 😉(sometimes using regex is easier for more complicated cases). Let me know which one you prefer 😄

Reply:
I didn't know how to make `vote_type` with paths, I'd use it otherwise.